internal/configs/transportserver.go

```diff
@@ -93,15 +94,18 @@ func generateUnixSocket(transportServerEx *TransportServerEx) string {
 	return ""
 }
 
-func generateStreamUpstreams(transportServerEx *TransportServerEx, upstreamNamer *upstreamNamer, isPlus bool) []version2.StreamUpstream {
+func generateStreamUpstreamsAndHealthCheck(transportServerEx *TransportServerEx, upstreamNamer *upstreamNamer, isPlus bool) ([]version2.StreamUpstream, *version2.StreamHealthCheck) {
 	var upstreams []version2.StreamUpstream
+	var hc *version2.StreamHealthCheck
 
 	for _, u := range transportServerEx.TransportServer.Spec.Upstreams {
```

Comment:
Can we remove the health check generation from this method?

At the moment, it appears to create a healthcheck for each upstream, but there is only one pointer. I think the value is overridden so it only returns the healthcheck for the last upstream, is this correct?

How about a new function, which takes the upstreams (transportServerEx.TransportServer.Spec.Upstreams ?) and returns the health checks for each one?

Reply:
> At the moment, it appears to create a healthcheck for each upstream, but there is only one pointer. I think the value is overridden so it only returns the healthcheck for the last upstream, is this correct?
 
Yeah that's right, at the moment we don't support multiple health checks, so I'll refactor so that the only health check that's generated is the one that referenced the right upstream